include/deal.II/base/parameter_handler.h

```diff
@@ -125,6 +125,90 @@ namespace Patterns
      * assumption significantly, you can still overload this function.
      */
     virtual std::size_t memory_consumption () const;
+
+
+    /**
+     * Convert a compatible value type to a string. If the type is not
+     * compatible, an excepion ExcIncompatibleType is thrown.
```

Comment:
Can you state what "compatible" here means?

I think the easiest way to do this would be to require every pattern to have a local typedef `value_type`, and to require that `T==DerivedType::value_type`.